app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.java

```diff
@@ -52,13 +52,11 @@ public void onDataSetChanged() {
             calendarEvents = calendarController.getNextDaysFromDb(14, this.appWidgetID);
 
             // set isFirstOnDay flags
-            Calendar currentDate = Calendar.getInstance();
-            Date startDate = new Date();
-            startDate.setTime(0);
-            currentDate.setTime(startDate);
+            DateTime currentDate = new DateTime(0);
```

Comment:
Because that's what the old code did :see_no_evil: 

I think we need a smallest lower bound for all days...
We could probably refactor this to a proper SQL query, à la
```SQL
SELECT min(dateTime) from IntegratedCalendarEvent GROUP BY date
```

Reply:
Wouldn't `DateTime.now()` be more readable?

Reply:
Since both of you didn't understand the logic, I figured rewriting it might be beneficial for readability...

Even though at first I didn't want to change existing and tested logic, but oh well :man_shrugging:

Reply:
Cheers